spec/routes/api/cli/golden_files_spec.rb

```diff
@@ -0,0 +1,86 @@
+    expect(PostgresResource).to receive(:generate_uuid).and_return("dd0375a6-1c66-82d0-a5e8-af1e8527a8a2")
+    expect(PostgresFirewallRule).to receive(:generate_uuid).and_return("5a601238-b56e-8ecf-bbca-9e3e680812b8")
+    expect(PostgresMetricDestination).to receive(:generate_uuid).and_return("1682bcf3-37f8-81da-a1e4-7ac9f25d9200")
+    cli(%w[vm eu-central-h1/test-vm create a])
+    @vm = Vm.first
+    add_ipv4_to_vm(@vm, "128.0.0.1")
+    @vm.nics.first.update(private_ipv4: "10.67.141.133/32", private_ipv6: "fda0:d79a:93e7:d4fd:1c2::0/80")
+
+    expect(Config).to receive(:postgres_service_hostname).and_return("pg.example.com").at_least(:once)
+    @dns_zone = DnsZone.new
+    expect(Vm).to receive(:generate_ubid).and_return(UBID.parse("vma9rnygexga6jns6x3yj9a6b2"))
+    expect(Prog::Postgres::PostgresResourceNexus).to receive(:dns_zone).and_return(@dns_zone).at_least(:once)
+    expect(PrivateSubnet).to receive(:generate_ubid).and_return(UBID.parse("psnqtahcasrj1hn16kh1ygekmn"))
+    expect(Firewall).to receive(:generate_uuid).and_return("30a3eec9-afb5-81fc-bbb5-8691d252ef03")
+    expect(FirewallRule).to receive(:generate_uuid).and_return("305d838d-a3cd-85f8-aa08-9a66e71a5877", "5aa5b086-37bd-81f8-8d03-dd4b0e09a436", "20c360fa-bc06-8df8-b067-33f4a1ebdbbd", "2b628450-25bd-8df8-8b42-fb5cc5d01ad1", "da42e2ef-b5f1-8df8-966d-1387afb1b2f4")
+
+    cli(%w[pg eu-central-h1/test-pg create])
+    cli(%w[pg eu-central-h1/test-pg reset-superuser-password bar456FOO123])
+    cli(%w[pg eu-central-h1/test-pg add-metric-destination foo bar https://baz.example.com])
+
+    expect(Vm).to receive(:generate_ubid).and_return(UBID.parse("vmz7b0dxt40t4g7rnmag9hct7c")).at_least(:once)
+    expect(PrivateSubnet).to receive(:generate_ubid).and_return(UBID.parse("ps9a8v5tm1020qn73f0c7db0x7")).at_least(:once)
+    fw_uuids = %w[2b4ae5cf-1aac-8dfc-bc80-c87e3e381e10 f5e6cb31-e580-81fc-88d6-a379f13494bf].cycle
+    expect(Firewall).to receive(:generate_uuid).and_invoke(-> { fw_uuids.next }).at_least(:once)
+    fwr_uuids = %w[c81cb3d1-81bc-89f8-96c8-6b4e8d4375bd e3fdf2f2-2603-85f8-b0f9-3fc2c28636cd 99785615-0cc7-8df8-a937-4f1d26b620c8 168054f0-e069-89f8-b12a-e4b010cf47b5 6d590d38-5f88-89f8-b6d8-0a079e1c61b6].cycle
+    expect(FirewallRule).to receive(:generate_uuid).and_invoke(-> { fwr_uuids.next }).at_least(:once)
+    expect(PostgresResource).to receive(:generate_uuid).and_return("97eb0a77-7869-86d0-9dcb-a46416ddc5c9").at_least(:once)
+    expect(PostgresFirewallRule).to receive(:generate_uuid).and_return("6d674a31-e1c1-8ecf-b5ac-363abb5b9185").at_least(:once)
+    expect(PostgresMetricDestination).to receive(:generate_uuid).and_return("bc563e43-9b83-89da-b3ac-d38acc87fd63").at_least(:once)
+
+    Dir["#{golden_file_dir}/*.txt"].each do |f|
+      f = File.basename(f)
+      cmd = f.delete_suffix(".txt")
+      status, cmd = cmd.split("-", 2)
```

Comment:
Since you are stubbing everything out above it may not come to it, but I thought I would share a little from our experience just in case.

For us, we explicitly writing out code that calls the tests (rather than figuring it out from filenames). One of the things we do along with that in some cases is supply substitutions, which are applied to the file. We use that to remove sensitive (or unstable) data. I think that your use of stubbing should largely address that in it's own right, but there may be exceptions.

Just thought I would mention it for your consideration moving forward. Some of our challenge is that in the golang context where we are doing things, stubbing would be much harder at best.

Reply:
I guess we could switch things to specify the command lines in the tests.  That would probably make adding new tests easier at least.  I can look into doing that in the future.

Reply:
Yeah, I think that offers more flexibility (since you can then have distinct setup/teardown things, whether they are substitution related or not), but it's also not something that was needed yet so makes sense to defer. As you suggest it probably could make adding new ones easier too, and also could avoid the need to do escaping (since the commands could then just be ruby text strings that match what they would otherwise look like, vs filenames), which might also help make things a bit easier to grok, update and maintain.

Reply:
I refactored how golden file commands are handled, so they are pulled from file content and not filenames in https://github.com/ubicloud/ubicloud/pull/2842/commits/9328334f7444dba7ab7fd09eda639331fc5c2bf6